### PR TITLE
Drupal rivers configurations

### PR DESCRIPTION
**Pull request by Dantheman720**
@unibrew - I wasn't able to get the indexer running on staging for some reason. I pushed up the provider and the rest of the configurations (outside of the river) should be the same. http://dcp.stage.jboss.org/v2/rest/indexer/jbossdeveloper_book/_status. I'm sure I did something wrong here. Everything works fine locally, but since I don't have visibility to the dcp log files, I don't really know what I did wrong here.

Anyway, perhaps we can discuss early next week?

### CODE REVIEW

**Comment by unibrew** [comment]
It seems that your push of provider was not successful, please retry, if still wrong, let me know I'll try myself - http://dcp.stage.jboss.org/v2/rest/provider/jboss-developer

File: rivers/books_drupal.json
Comment (unibrew) [inline rivers/books_drupal.json]: I think you've reimplemented in JS this preprocessor https://github.com/searchisko/structured-content-tools/blob/master/src/main/java/org/jboss/elasticsearch/tools/content/LongToTimestampValuePreprocessor.java

File: rivers/books_drupal.json
Comment (unibrew) [inline rivers/books_drupal.json]: Not sure if intentionally you prep sys authors but then put them in authors field instead of sys_authors.

File: rivers/books_drupal.json
Comment (unibrew) [inline rivers/books_drupal.json]: This field seems redundant to sys_created, plus 'test_sys_created_date' doesn't exist.

**Comment by Dantheman720** [comment]
@unibrew - So, I think this is about ready. Can we go through it tomorrow and discuss the sys_contributors further? I wasn't able to look into it much today, but I've been hung up on it for quite of a bit.

For the LongToTimestamp conversion, it wasn't actually being converted to a timestamp so I decided to use JS to parse it. If that doesn't work, I can definitely change it.

**Comment by Dantheman720** [comment]
@unibrew - Ok! I think we're good to go!

**Comment by unibrew** [comment]
Is the same on our stage? I'd like to take the last look on the data before merging tomorrow.

**Comment by Dantheman720** [comment]
I'll push each of those up to staging. Though...we don't have the books and connectors in production on the drupal side yet. I need to get those in before we merge otherwise it'll just pull empty data sets. I'll do that first thing tomorrow morning.

**Comment by unibrew** [comment]
ok, understood. We'll see how it works on staging but I'm waiting for your green light till this gets pushed to prod. Thanks!

**Comment by Dantheman720** [comment]
Sounds like a plan!

**Comment by Dantheman720** [comment]
Well, staging is up to date now. I'll work on production tomorrow.

**Comment by unibrew** [comment]
This space key issue that Libor raised against SO can possibly be present here as well. We need to check it.

**Comment by Dantheman720** [comment]
@unibrew - So, I think we're good now? I pushed the new configs to staging.

**Comment by Dantheman720** [comment]
I haven't moved anything to production yet, since I'm not entirely sure this is what's required.

**Comment by unibrew** [comment]
Thank you, I'll take a final look on the data on staging on Monday. Is everything production ready on Drupal side? Last time you've mentioned that some work needs to be done before pushing this config to prod.
Also, would you mind sending me an email with the command that will add the correct configuration for apikey for prod. I suppose you have it cached somewhere, I wouldn't have to "invent" it ;)

**Comment by Dantheman720** [comment]
I need https://github.com/redhat-developer/developers.redhat.com/pull/1756 merged before it's prod ready.

**Comment by unibrew** [comment]
Dan, you need to have a look on errors /v2/rest/indexer/_all/_status. There are date format issues for sys_created. I stopped those 3 rivers so that they don't spam logs with those errors. Remember to restart them after correcting the config.

**Comment by Dantheman720** [comment]
Will do!

**Comment by unibrew** [comment]
@lkrzyzanek or @velias : guys, any of you want to review this as well before I merge?

**Comment by velias** [comment]
I miss prefix jbossdeveloper_ in river names to keep naming conventions

**Comment by Dantheman720** [comment]
I think we should be good to go now. The data is in production for books and connectors. We are not pushing old events.

**Comment by lkrzyzanek** [comment]
Yep I can review. Where is JIRA issue for this with description what it suppose to do?

**Comment by Dantheman720** [comment]
This was the JIRA ticket I've been working from -
https://issues.jboss.org/browse/DEVELOPER-3809. It doesn't really get into
the technical requirements of the rivers config, but it defines our
requirements for RHD.

**Comment by lkrzyzanek** [comment]
Review jira issue: https://issues.jboss.org/browse/ORG-3616

**Review by lkrzyzanek — CHANGES_REQUESTED** [review]
1. General comments 
 * `urlGetDocuments` contains http:// I would suggest to move http to the configuration. It would be more flexible in case of https change
 * rivers uses copying data from source field to "prep_something" in cases  where it's not necessary to do so. When you don't need to manipulate data in preprocessor then directly copy data from remote field within index.fields definition. It would keep things more cleaner. 
* timeout - 20secs. Is production fast enough? Consider increasing to 60 sec
* all mappings "data_something" doesn't have mapping for `space_key`
````
"space_key" : {"type" : "string", "analyzer" : "keyword"},
````

2. Mapping for Books - mappings/data_jbossdeveloper_book/jbossdeveloper_book.json 
* needs mapping for `published_date` as date

3. Mapping for Connector mappings/data_jbossdeveloper_connector/jbossdeveloper_connector.json
* priority needs mapping to be `integer` otherwise sorting doesn't work properly
* I'm also wondering what the value of priority means. 50 > 10 ? In that case registered query sorts based on it incorrectly

4. Events
I'm not sure if field_data_tags works correctly. Source contains string like "Atomic, CDK, Containers". Check if ES parses it to the array out of the box.

**Comment by lkrzyzanek** [comment]
Btw. books doesn't have relation to product so I raised this issue DEVELOPER-3915. But it can be fixed in separate PR.

**Comment by Dantheman720** [comment]
@lkrzyzanek - "urlGetDocuments contains http:// I would suggest to move http to the configuration. It would be more flexible in case of https change."

I tried leaving the http:// out and for some reason, I kept getting error messages. I guess that could've been related to parsing.  Other than that, I'll make the suggested changes.

File: mappings/data_jbossdeveloper_connector/jbossdeveloper_connector.json
Comment (lkrzyzanek) [inline mappings/data_jbossdeveloper_connector/jbossdeveloper_connector.json]: The final property is `priority` not `field_connector_priority`

**Review by lkrzyzanek — COMMENTED** [review]
I think it looks good now. Try those rivers in  dcp stage and I'll do final review of final content in stage.
Thanks.

**Comment by Dantheman720** [comment]
Good deal. I'll make that switch. I wasn't 100% sure on that.

**Comment by Dantheman720** [comment]
@lkrzyzanek - Alright, everything is in staging now (including the fix). Hopefully we're good to merge now.

**Review by lkrzyzanek — APPROVED** [review]

**Comment by lkrzyzanek** [comment]
Looks good to me know. I think we can merge.
What is the release steps in PROD ?

**Comment by Dantheman720** [comment]
@unibrew - I think we're good now! Do you need me to do anything for the release?

**Comment by unibrew** [comment]
Guys, I'm on PTO till Monday and that's the earliest I can look into this. Unless Libor handles this before.

**Comment by lkrzyzanek** [comment]
@Dantheman720 what are steps you did on staging. We should do same in PROD. I'm looking for something like this: https://issues.jboss.org/browse/ORG-3611?focusedCommentId=13363857&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13363857

**Comment by Dantheman720** [comment]
@lkrzyzanek - I haven't worked with production before. I basically just blew everything out for each index and river.  The list that you provided on https://issues.jboss.org/browse/ORG-3611?focusedCommentId=13363857&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel looks pretty similar to what I did (actually it's more elegant than what I did).

**Comment by lkrzyzanek** [comment]
Ok, I've merged the PR but we would still need exact release steps.
@Dantheman720 can you work on this? I mean just list of commands needed.
I guess it will be very similar what I used before so feel free to copy :-)

**Comment by Dantheman720** [comment]
@unibrew - Do you need me to do anything here? I've not done anything related to release before.

**Comment by unibrew** [comment]
@Dantheman720 Please have a look if I didn't miss anything in the release steps https://issues.jboss.org/browse/ORG-3626

**Comment by unibrew** [comment]
This is now pushed to prod. Thank you all for the PR and reviews.

**Pull request merged**
